### ISSUE TITLE
Module separation

### ISSUE THREAD

**Issue by lilgallon** [IZIVIA/ocpi-toolkit#111]
What is the purpose of having separate `ocpi-2-2-1` and `ocpi-transport` maven dependencies?

`ocpi-2-2-1` should include `ocpi-transport` in its jar. That way, users of the lib will just need to import `ocpi-2-2-1` in their project.

Later, we could also provide transport implementations, and it would make sense in that case to create separate jars, because they will be optional (example: `ocpi-transport-http4k-impl`)

**Comment by atschabu** [comment]
I guess it made more sense while there was a ocpi 2.1 implementation. It might be relevant again when there is a 3.x implementation. But the relation is not as clear cut, I have far too many "hacks" to get the transport layer work nicely with ocpi module

**Comment by lilgallon** [comment]
Yes it was made because the transport layer is common for each OCPI version. The thing is that it should not be a separate maven dependency because it's not optional, but mandatory (main ocpi module depends on the transport module)

I am not super satisfied with the way the transport part of ocpi module is designed either. It may change in the future, but it's not the priority at the moment
